### PR TITLE
Add admin overview page

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Replaces the tiles in "Admin Portals" with a new view called "Overview" as a step toward a more complete overhaul of the UI.

## Related Tickets & Documents

Solves RFC: https://github.com/forem/rfcs/pull/39/files

## QA Instructions, Screenshots, Recordings

[Mockup Screenshot](https://user-images.githubusercontent.com/816402/105643299-e6a00a80-5e43-11eb-98e6-9380f489d791.jpg)

After the changes in this PR:
image

(A few of the links from the mockup were removed during this PR)

### UI accessibility concerns?

~Axe isn't happy with the color contrast.~ **FIXED**
image

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/views/admin/overview/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<h2 class="crayons-title mb-3">Overview</h2>
+<div class="grid s:grid-cols-3 s:grid-rows-4 gap-3">
+  <article style="grid-column: span 2; grid-row: span 4;">
```
Comment (jacobherrington) [inline app/views/admin/overview/index.html.erb:3]: I'd love to get rid of these inline styles, but I'm not sure if we have utility classes to handle these attributes. Does anyone know if we do?

File: app/views/admin/overview/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          <a role="button" href="https://creators.forem.com" class="crayons-btn mt-1 w-100 self-end">Join Community</a>
+        </div>
+      </section>
+      <section class="crayons-card">
+        <div class="crayons-card__header">
+          <h3 class="crayons-subtitle-2">Forem Community Experts</h3>
+        </div>
+        <div class="crayons-card__body flex flex-wrap" style="min-height: 150px;">
+          <p>Get one-on-one time with our expert who can give you an extra hand.</p>
+          <a role="button" href="" class="crayons-btn mt-1 w-100 self-end">Book Time</a>
```
Comment (jacobherrington) [inline app/views/admin/overview/index.html.erb:47]: I don't have a link here, does anyone know what this should be?

cc: @vaidehijoshi @pkfrank

File: docs/getting-started/start-app.md [hunk 3]
```diff
@@ -53,9 +53,9 @@ advanced setup to start the application by using [Overmind][overmind].
 The prerequisite is to install [Overmind][overmind], a process manager which
 takes advantage of [tmux][tmux].
 
-This will allow you to launch all your app's processes in the same terminal, navigate the logs
-of each service separately, restart each service separately and have a better
-debugging experience.
+This will allow you to launch all your app's processes in the same terminal,
+navigate the logs of each service separately, restart each service separately
+and have a better debugging experience.
```
Comment (jacobherrington) [inline docs/getting-started/start-app.md:58]: Most of the diff in this file is noise from prettier.

File: app/views/admin/overview/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          <a role="button" href="https://creators.forem.com" class="crayons-btn mt-1 w-100 self-end">Join Community</a>
+        </div>
+      </section>
+      <section class="crayons-card">
+        <div class="crayons-card__header">
+          <h3 class="crayons-subtitle-2">Forem Community Experts</h3>
+        </div>
+        <div class="crayons-card__body flex flex-wrap" style="min-height: 150px;">
+          <p>Contact support to set up a free platform training session with a Forem expert.</p>
+          <a role="button" href="mailto:support@forem.com" class="crayons-btn mt-1 w-100 self-end">Contact Support</a>
```
Comment (jacobherrington) [inline app/views/admin/overview/index.html.erb:47]: Based on Peter's suggestion we should just set these up manually with the community folks.

The UI is a bit different from the mock here:
image

**Comment by jacobherrington** [comment]
@vaidehijoshi @mstruve I haven't address tracking interactions with the UI in Ahoy. That should be a reasonably trivial task, but I would love some guidance on how that data should be structured.

As I understand it, our motivation for tracking here is to confirm that people use these links at all, would it make the most sense to use a single event that verifies people have clicked a link on this page (excluding the header and sidebar navigation)? Or should we be tracking each link on the page individually?

I will also need to configure Ahoy.js in the admin UI as it's currently only being used in views outside of the admin interface, but that should be just a few lines of code.

**Comment by mstruve** [comment]
I would vote for tracking the individual links explicitly that way if we want we can break the data down, if we dont we can combine and consolidate it into a single bucket. At least with the additional detail, we have the option.

File: app/views/admin/overview/index.html.erb (hunk 4 above)
Comment (vaidehijoshi) [reply app/views/admin/overview/index.html.erb:47]: > we should just set these up manually with the community folks.

To confirm, this means that we don't need the user (the Forem Creator) to take any action then, right? The action would instead come from our side, the Forem bizdev side?

If that's right, then I think we just remove this bit from the design. When we came up with the design mocks in the original RFC (https://github.com/forem/rfcs/pull/39), it was under the assumption that all these links were ones we wanted to include; if that has changed, I think we can just remove this from the design given that we don't have anywhere to link to :)

**Comment by vaidehijoshi** [comment]
> As I understand it, our motivation for tracking here is to confirm that people use these links at all, would it make the most sense to use a single event that verifies people have clicked a link on this page (excluding the header and sidebar navigation)? Or should we be tracking each link on the page individually?
> 
> I will also need to configure Ahoy.js in the admin UI as it's currently only being used in views outside of the admin interface, but that should be just a few lines of code.

We should definitely track the links on the page individually since the intention of adding tracking here is for us to see what resources Forem Creators are actually clicking on, and the extent to which  they find this new overview page helpful or not.

To that end, we should make it easy to see (on each Forem) whether the Forem admins are actually clicking these links, and _which ones_ they're using (hence the reason for individual tracking).

You might be able to follow some of what I did on [the notifications page](https://github.com/forem/forem/blob/ca6ba1af7d7fa7f583262381d586e032f4aac211/app/views/notifications/index.html.erb#L38) when I set up when collecting data from welcome notifications on the notification view (admittedly, this is not on the admin UI so you'll still have to port that into the admin portal!)

File: app/views/admin/overview/index.html.erb (hunk 4 above)
Comment (vaidehijoshi) [reply app/views/admin/overview/index.html.erb:47]: @jacobherrington let me know if you are blocked on this at all; i just want to make sure that you aren't waiting on any info for these links :)

**Comment by jacobherrington** [comment]
Had a bit of an environment issue with ahoy that ate up some time, but this should be ready for review now! :+1:

**Comment by jacobherrington** [comment]
Currently, when a user clicks on one of these links we make an event that looks like this:
```ruby
#<Ahoy::Event:0x000000000e0b9ab0
 id: 1,
 name: "Overview Link Clicked",
 properties: {"action"=>"click", "target"=>"http://localhost:3000/admin/invitations"},
 time: Fri, 26 Feb 2021 19:26:23 UTC +00:00,
 user_id: 11,
 visit_id: 1>
 ```
 
 Does that seem like enough information @vaidehijoshi or is there more we should be tracking here? I feel like this is enough for us to have a good idea of which links are being clicked, but it could definitely be more granular if that's not enough.

File: app/javascript/admin/controllers/ahoy_controller.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+// eslint-disable-next-line no-restricted-syntax
+export default class AhoyController extends Controller {
+  trackOverviewLink(event) {
+    event.preventDefault();
+    let properties = {
+      action: event.type,
+      target: event.target.toString(),
+    };
+    ahoy.track('Overview Link Clicked', properties);
```
Comment (vaidehijoshi) [inline app/javascript/admin/controllers/ahoy_controller.js:17]:
```suggestion
    ahoy.track('Admin Overview Link Clicked', properties);
```
Might just make it a tad bit easier down the road when we want to parse all events by the admin portal (as an example)

File: app/views/admin/overview/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,54 @@
+<h2 class="crayons-title mb-3">Overview</h2>
+<div class="grid s:grid-cols-3 s:grid-rows-4 gap-3">
+  <article style="grid-column: span 2; grid-row: span 4;">
+    <section class="crayons-card mb-3">
+      <div class="crayons-card__body" data-controller="ahoy">
+        <h3 class="crayons-subtitle-2">Welcome Checklist</h3>
+        <p class="crayons-field__description mb-3">Here are things we recommend you do now that you're on Forem</p>
```
Comment (vaidehijoshi) [inline app/views/admin/overview/index.html.erb:7]:
```suggestion
        <p class="crayons-field__description mb-3">Here are a few things we recommend doing now that you're using Forem!</p>
```

I know the design mock has different text, but when I was playing with it locally, this just read a little funky to me. What do you think about this instead?

File: app/views/admin/overview/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        <ol>
+          <li>Setup and configure your Forem under <a href="/admin/config" data-action="click->ahoy#trackOverviewLink">Config</a></li>
```
Comment (vaidehijoshi) [inline app/views/admin/overview/index.html.erb:9]:
```suggestion
          <li>Set up and configure your Forem under <a href="/admin/config" data-action="click->ahoy#trackOverviewLink">Config</a></li>
```

File: app/views/admin/overview/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+          <li><a href="/admin/invitations" data-action="click->ahoy#trackOverviewLink">Invite people</a> to join your Forem</li>
+          <li><a href="/admin/tags" data-action="click->ahoy#trackOverviewLink">Update your Forem tags</a></li>
+          <li><a href="/admin/pages" data-action="click->ahoy#trackOverviewLink">Create pages</a> and <a href="/admin/navigation_links" data-action="click->ahoy#trackOverviewLink">sort them in the navigation sidebar</a></li>
+          <li>Read the <a href="https://forem.gitbook.io/forem-admin-guide/quick-start-guide" data-action="click->ahoy#trackOverviewLink">Quick Start Guide</a></li>
+          <li><a href="/admin/welcome" data-action="click->ahoy#trackOverviewLink">Set up a Welcome Thread</a></li>
+          <li>Browse the <a href="https://forem.gitbook.io/forem-admin-guide" data-action="click->ahoy#trackOverviewLink">Forem Admin Guide</a></li>
+          <li>Join the private <a href="https://creators.forem.com">Creators Forem</a>! (email <a href="mailto:support@forem.com" data-action="click->ahoy#trackOverviewLink">support@forem.com</a> if you need an invite)</li>
```
Comment (vaidehijoshi) [inline app/views/admin/overview/index.html.erb:16]:
```suggestion
          <li>Join the private <a href="https://creators.forem.com">Creators Forem</a> (email <a href="mailto:support@forem.com" data-action="click->ahoy#trackOverviewLink">support@forem.com</a> if you need an invite)</li>
```
Just to keep it consistent with the other items in the list don't have punctuation :)

**Review by vaidehijoshi — COMMENTED** [review]
Just some super minor comments!

**Comment by vaidehijoshi** [comment]
> Axe isn't happy with the color contrast.

Do we need to use `color-base-40` here? It seems like we don't use it anywhere else, so if that's the element in question, maybe we can use something else here and fix up the contrast while we're in the code 🤔

File: app/views/admin/overview/index.html.erb (hunk 1 above)
Comment (citizen428) [reply app/views/admin/overview/index.html.erb:3]: @ludwiczakpawel probably knows the answer to this 😃

Comment (ludwiczakpawel) [reply app/views/admin/overview/index.html.erb:3]: We do not have utility classes for these. We can either add these to the [generator](https://github.com/forem/forem/blob/master/app/assets/stylesheets/config/_generator.scss) (I'm happy to help) or move these particular styles to appropriate CSS file oooor just leave it as you have here. I do think it's ugly and naughty :D but maybe it's just not worth the effort IF we have any plans to revisit this view anytime soon?

**Comment by ludwiczakpawel** [comment]
> Axe isn't happy with the color contrast.

I don't think it's strictly related to this PR so you can either ignore it, especially it's just for _decoration_ element (`»`) or you can replace `color-base-40` with `color-base-60` - this is the lightest shade of grey that passes contrast ratio test (afair).

ooooooooooooor (and please don't take it too serious) you can trick axe and ppl with visual impairment with `color-base-60 opacity-50` (that's a very bad idea, don't listen to me).

File: app/views/admin/overview/index.html.erb (hunk 1 above)
Comment (ludwiczakpawel) [reply app/views/admin/overview/index.html.erb:3]: ohhh, wait a minute....

here's something you may find useful:

```
<h2 class="crayons-title mb-3">Overview</h2>
<div class="px-0 crayons-layout crayons-layout--2-cols crayons-layout--2-cols--1-2 crayons-layout--2-cols--inverted">
    <div class="crayons-layout__content">
        Left column
    </div>
    <div class="crayons-layout__sidebar-right">
        Right column
    </div>
</div>
```

**Review by vaidehijoshi — APPROVED** [review]
this will be a great change to share with our forem creators :)

**Review by maestromac — APPROVED** [review]
LGTM!

**Pull request merged**